### PR TITLE
Switch deployment workflow to re-usable action.

### PR DESCRIPTION
**Pull request by IAlibay**
This switches the deployment to a re-usable action for a little bit more consistency across our ecosystem, and also some additional tests.

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -8,55 +8,27 @@ on:
...
+      - name: testpypi_deploy
+        uses: MDAnalysis/pypi-deployment@main
         if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/')
-        uses: pypa/gh-action-pypi-publish@release/v1
         with:
-          password: ${{ secrets.TEST_PYPI_API_TOKEN }}
-          repository_url: https://test.pypi.org/legacy/
+          token: ${{ secrets.TEST_PYPI_API_TOKEN }}
+          test_submission: true
+          package_name: 'gridData'
```
Comment (lilyminium) [inline .github/workflows/deploy.yaml:27]: Isn't this gridDataFormats?

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -8,55 +8,27 @@ on:
...
 
-      - name: publish_pypi
+      - name: pypi_deploy
+        uses: MDAnalysis/pypi-deployment@main
         if: github.event_name == 'release' && github.event.action == 'published'
-        uses: pypa/gh-action-pypi-publish@release/v1
         with:
-          password: ${{ secrets.PYPI_API_TOKEN }}
+          token: ${{ secrets.PYPI_API_TOKEN }}
+          package_name: 'gridData'
```
Comment (lilyminium) [inline .github/workflows/deploy.yaml:34]: Same question above -- this variable is used for `pip install`, right?

Comment (IAlibay) [reply .github/workflows/deploy.yaml:34]: No it's used for the `pytest --pyargs` call, I _think_ this is correct.

Comment (IAlibay) [reply .github/workflows/deploy.yaml:34]: ah wait I messed up somewhere

Comment (IAlibay) [reply .github/workflows/deploy.yaml:34]: blergh what a pain - I'll have to change the action to account for this... funilly enough it didn't trip up mda_xdrlib at all.

Comment (IAlibay) [reply .github/workflows/deploy.yaml:34]: Huh I guess I never fully realised pip doesn't care between `-` and `_` in package names.

Comment (lilyminium) [reply .github/workflows/deploy.yaml:34]: Ah, I can't believe I keep spending the effort on typing underscores now 😂 better safe than sorry I guess

**Review by orbeckst — APPROVED** [review]
Let's try it.

**Pull request merged**